### PR TITLE
Add doc for "How to compile the Openresty" and fix the doc about how to install apisix.

### LINKED ISSUE #830: request help: Add document about how to compile the Openresty.
### Issue description
There is no doc in "install-dependencies.md" about how to compile the Openresty

### Environment

* apisix version (cmd: `apisix version`): 0.8 or higher
* OS: tlinux1.2/tlinux2.2

### PR DESCRIPTION
**Pull request by Miss-you**
NOTE: Please read the Contributing.md guidelines before submitting your patch:

https://github.com/apache/incubator-apisix/blob/master/Contributing.md#how-to-add-a-new-feature-or-change-an-existing-one

### Summary

SUMMARY_HERE

### Full changelog

* add doc for "How to compile the Openresty"

### Issues resolved

Fix #830

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ The main steps to install APISIX:
 ```shell
 sudo yum install yum-utils
 sudo yum-config-manager --add-repo https://openresty.org/package/centos/openresty.repo
-sudo yum install -y openresty etcd
+sudo yum install -y openresty etcd lua-devel
```
Comment (membphis) [inline README.md]: I do not think we need `lua-devel` now.

File: doc/install-dependencies.md [hunk 2]
```diff
@@ -24,6 +24,7 @@
 * [Ubuntu 16.04 & 18.04](#ubuntu-1604--1804)
 * [Debian 9 & 10](#debian-9--10)
 * [Mac OSX](#mac-osx)
+* [How to compile the Openresty](#how-to-compile-the-openresty)
```
Comment (moonming) [inline doc/install-dependencies.md]: better link to openresty install doc

File: doc/install-dependencies.md [hunk 3]
```diff
@@ -19,11 +19,13 @@
 
 # Install Dependencies
 
-* [CentOS 6](#centos-6)
-* [CentOS 7](#centos-7)
-* [Ubuntu 16.04 & 18.04](#ubuntu-1604--1804)
-* [Debian 9 & 10](#debian-9--10)
-* [Mac OSX](#mac-osx)
+- [Install Dependencies](#install-dependencies)
```
Comment (moonming) [inline doc/install-dependencies.md]: No content for this link?

File: doc/install-dependencies.md [hunk 4]
```diff
@@ -120,3 +122,29 @@ brew install openresty/brew/openresty etcd luarocks curl git
 # start etcd server with v2 protocol
 etcd --enable-v2=true &
 ```
+
+How to compile the Openresty
```
Comment (moonming) [inline doc/install-dependencies.md]: Compiling OpenResty from source is very complicated, it's not easy to make it clear. So I don't think we need to explain how to compile OpenResty, just a link is enough.

Comment (membphis) [reply doc/install-dependencies.md]: Agree with this.

File: doc/install-dependencies.md (hunk 3 above)
Comment (membphis) [reply doc/install-dependencies.md]: same question

**Comment by membphis** [comment]
@Miss-you many thx, merged

**Pull request merged**